foundry/foundry.py

```diff
@@ -155,38 +154,24 @@ def __init__(
             force_login=False,
         )
 
-        if name is not None:
-            self._load(name=name,
-                       download=download,
-                       globus=globus,
-                       verbose=verbose,
-                       metadata=metadata,
-                       authorizers=authorizers,
-                       interval=interval)
+        # handle empty dataset name (was returning all the datasets)
```

Comment:
I'm not sure I understand this comment, could you give me a little more context?

Reply:
This comment got kicked around and ended up in a weird place - I put it back in the `download_metadata` with some logic to ensure it wasn't called without a name argument, which appears to download all the datas.